src/sentry/api/endpoints/organization_event_details.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+import six
+
+from sentry.api.bases import OrganizationEventsEndpointBase, OrganizationEventsError, NoProjects
+from sentry import features
+from sentry.models import SnubaEvent
+from sentry.models.project import Project
+from sentry.api.serializers import serialize
+from sentry.utils.snuba import raw_query
+
+
+class OrganizationEventDetailsEndpoint(OrganizationEventsEndpointBase):
+    def get(self, request, organization, project_slug, event_id):
+        if not features.has('organizations:events-v2', organization, actor=request.user):
+            return Response(status=404)
+
+        try:
+            params = self.get_filter_params(request, organization)
+            snuba_args = self.get_snuba_query_args_v2(request, organization, params)
+        except OrganizationEventsError as exc:
+            return Response({'detail': exc.message}, status=400)
+        except NoProjects:
+            return Response(status=404)
+
+        try:
+            project = Project.objects.get(
+                slug=project_slug,
+                organization_id=organization.id
+            )
+        except Project.DoesNotExist:
+            return Response(status=404)
+
+        # We return the requested event if we find a match regardless of whether
+        # it occurred within the range specified
+        event = SnubaEvent.objects.from_event_id(event_id, project.id)
+
+        if event is None:
+            return Response({'detail': 'Event not found'}, status=404)
+
+        data = serialize(event)
+
+        data['nextEventID'] = self.next_event_id(request, organization, snuba_args, event)
+        data['previousEventID'] = self.prev_event_id(request, organization, snuba_args, event)
+        data['projectSlug'] = project_slug
+
+        return Response(data)
+
+
+class OrganizationEventDetailsLatestEndpoint(OrganizationEventsEndpointBase):
```

Comment:
These two endpoints could be abstracted too.

Reply:
Updated latest/oldest to share code